Remove MeshHandle (#608)
precice/precice: CHANGELOG.md

## develop

> - Remove `MeshHandle` and replace only valid use by adding `SolverInterfaceImpl::mesh()`.

Reviewer: ```suggestion
- Remove `MeshHandle` from API and replace use in integration tests by `SolverInterfaceImpl::mesh()`.
```
